---
title: "Survival Analysis with R"
---

This workshop will provide hands-on instruction and exercises covering survival analysis using R. The data to be used here will come from The Cancer Genome Atlas (TCGA), where we may also cover programmatic access to TCGA through Bioconductor if time allows. 

**Prerequisites: [Familiarity with R](r-basics.html) is _required_** (including working with [data frames](r-dataframes.html), installing/using packages, importing data, and saving results); familiarity with [dplyr](r-dplyr-yeast.html) and [ggplot2](r-viz-gapminder.html) packages is highly recommended. 

**You must [complete the basic R setup here](setup.html#R) _prior to class_.** This includes installing R, RStudio, and the required packages. Please [contact one of the instructors](people.html) _prior to class_ if you are having difficulty with any of the setup. Please bring your laptop and charger cable to class.

<!--
**Handouts**: Download and print out these handouts and bring them to class:

- [Cheat sheet](handouts/r-stats-cheatsheet.pdf)
- [Exercises handout](handouts/r-stats-exercises.pdf)
-->

```{r, echo=FALSE, message=FALSE, eval=TRUE}
library(knitr)
opts_chunk$set(message=FALSE, warning=FALSE, eval=TRUE, echo=TRUE, cache=FALSE)
# options(digits=3)
options(max.print=200)
.ex <- 1 # Track ex numbers w/ hidden var. Increment each ex: `r .ex``r .ex=.ex+1`
```

## Background

In the class on [essential statistics](r-stats.html) we covered basic categorical data analysis -- comparing proportions (risks, rates, etc) between different groups using a chi-square or fisher exact test, or logistic regression. For example, we looked at how the diabetes rate differed between males and females. In this kind of analysis you implicitly assume that the rates are constant over the period of the study, or as defined by the different groups you defined. 

But, in longitudinal studies where you track samples or subjects from one time point (e.g., entry into a study, diagnosis, start of a treatment) until you observe some outcome _event_ (e.g., death, onset of disease, relapse), it doesn't make sense to assume the rates are constant. For example: the risk of death after heart surgery is highest immediately post-op, decreases as the patient recovers, then rises slowly again as the patient ages. Or, recurrence rate of different cancers varies highly over time, and depends on tumor genetics, treatment, and other environmental factors. 

### Definitions

**Survival analysis** lets you analyze the rates of occurrence of events over time, without assuming the rates are constant. Generally, survival analysis lets you model the _time until an event occurs_,[^deathvs] or compare the time-to-event between different groups, or how time-to-event correlates with quantitative variables.

[^deathvs]: In the medical world, we typically think of _survival analysis_ literally -- tracking time until death. But, it's more general than that -- survival analysis models time until an _event_ occurs (_any_ event). This might be death of a biological organism. But it could also be the time until a hardware failure in a mechanical system, time until recovery, time someone remains unemployed after losing a job, time until a ripe tomato is eaten by a grazing deer, time until someone falls asleep in a workshop, etc. _Survival analysis_ also goes by _reliability theory_ in engineering, _duration analysis_ in economics, and _event history analysis_ in sociology.

The **hazard** is the instantaneous event (death) rate at a particular time point _t_. Survival analysis doesn't assume the hazard is constant over time. The _cumulative hazard_ is the total hazard experienced up to time _t_.

The **survival function**, is the probability an individual survives (or, the probability that the event of interest does not occur) up to and including time _t_. It's the probability that the event (e.g., death) hasn't occured yet. It looks like this, where $T$ is the time of death, and $Pr(T>t)$ is the probability that the time of death is greater than some time $t$. $S$ is a probability, so $0 \leq S(t) \leq 1$, since survival times are always positive ($T \geq 0$).

$$ S(t) = Pr(T>t) $$


The **Kaplan-Meier** curve illustrates the survival function. It's a step function illustrating the cumulative survival probability over time. The curve is horizontal over periods where no event occurs, then drops vertically corresponding to a change in the survival function at each time an event occurs. 

**Censoring** is a type of missing data problem unique to survival analysis. This happens when you track the sample/subject through the end of the study and the event never occurs. This could also happen due to the sample/subject dropping out of the study for reasons other than death, or some other loss to followup. The sample is _censored_ in that you only know that the individual survived up to the loss to followup, but you don't know anything about survival after that.[^censoring]

[^censoring]: This describes the most common type of censoring -- _right censoring_. _Left censoring_ less commonly occurs when the "start" is unknown, such as when an initial diagnosis or exposure time is unknown. 

**Proportional hazards assumption**: The main goal of survival analysis is to compare the survival functions in different groups, e.g., leukemia patients as compared to cancer-free controls. If you followed both groups until everyone died, both survival curves would end at 0%, but one group might have survived on average a lot longer than the other group. Survival analysis does this by comparing the _hazard_ at different times over the observation period. Survival analysis doesn't assume that the hazard is constant, but _does_ assume that the _ratio_ of hazards between groups is constant over time.[^cumhaz] This workshop does _not_ cover methods to deal with non-proportional hazards, or interactions of covariates with the time to event.

[^cumhaz]: And, following the definitions above, assumes that the _cumulative hazard_ ratio between two groups remains constant over time.

**Proportional hazards regression** a.k.a. **Cox regression** is the most common approach to assess the effect of different variables on survival. 

### Cox PH Model

Kaplan-Meier curves are good for visualizing differences in survival between two categorical groups,[^logrank] but they don't work well for assessing the effect of _quantitative_ variables like age, gene expression, leukocyte count, etc. Cox PH regression can assess the effect of both categorical and continuous variables, and can model the effect of multiple variables at once.[^multregression]

[^logrank]: And there's a chi-square-like statistical test for these differences called the [log-rank test](https://en.wikipedia.org/wiki/Log-rank_test) that compare the survival functions categorical groups.

[^multregression]: See the [multiple regression](r-stats.html#multiple_regression) section of the essential statistics lesson.

Cox PH regression models the natural log of the hazard at time _t_, denoted $h(t)$, as a function of the baseline hazard ($h_0(t)$) (the hazard for an individual where all exposure variables are 0) and multiple exposure variables $x_1$, $x_1$, $...$, $x_p$. The form of the Cox PH model is:

$$ log(h(t)) = log(h_0(t)) + \beta_1 x_1 + \beta_2 x_2 + ... + \beta_p x_p $$

If you exponentiate both sides of the equation, and limit the right hand side to just a single categorical exposure variable ($x_1$) with two groups ($x_1=1$ for exposed and $x_1=0$ for unexposed), the equation becomes:

$$ h_1(t) = h_0(t) + e^{\beta_1 x_1} $$ 

Rearranging that equation lets you estimate the **hazard ratio**, comparing the exposed to the unexposed individuals at time _t_: 

$$ HR(t) = \frac{h_0(t) e^{\beta_1}}{h_0(t)} = e^{\beta_1} $$

This model shows that **the hazard ratio is $e^{\beta_1}$,** and remains constant over time _t_ (hence the name _proportional hazards regression_). The $\beta$ values are the regression coefficients that are estimated from the model, and represent the $log hazard ratio$ for each unit increase in the corresponding predictor variable. The interpretation of the hazards ratio depends on the measurement scale of the predictor variable, but in simple terms, a positive coefficient indicates worse survival and a negative coefficient indicates better survival for the variable in question.

## R

Use the **survival** package. Built in, must load. 

- `Surv()`: Creates a survival object.
- `survfit()`: Fits a survival curve using either a formula, of from a previously fitted Cox model.
- `coxph()`: Fits a Cox proportional hazards regression model.
- `cox.zph()`: Tests the proportional hazards assumption of a Cox regression model.

`Surv()` creates the response variable, and typical usage takes the time to event[^time2], and whether or not the event occured (death vs censored). Models specified the same way as in regular linear models, but using the `coxph()` function.

[^time2]: `Surv()` can also take start and stop times, to account for left censoring. See the help for `?Surv`.

## Cox Regression

The response variable you create with `Surv()` goes on the left hand side of the formula, specified with a `~`. Explanatory variables go on the right side. Uses the same syntax as `lm()`, `glm()`, etc.

### Interpreting the output

The `exp(coef)` column contains $e^{\beta_1}$ (see [background](#background) section above for more info). This is the **hazard ratio** -- the multiplicative effect of that variable on the hazard rate (for each unit increase in that variable). So, for a categorical variable like sex, going from male (baseline) to female results in approximately 43% reduction in hazard. You could also flip the sign on the `coef` column, and take `exp(0.551)`, which you can interpret as being male resulting in a 1.73-fold increase in hazard, or that males die ad approximately 1.73x the rate per unit time as females (females die at 0.551x the rate per unit time as males). 

Just remember:

- HR=1: No effect
- HR>1: Increase in hazard
- HR<1: Reduction in hazard (protective)

```{r scratch, eval=FALSE, include=FALSE}
library(tidyverse)
library(survival)
library(survminer)
aml
lung

# https://plot.ly/ipython-notebooks/survival-analysis-r-vs-python/

# ?lung

# setting up the response variable
s <- Surv(time=lung$time, event=lung$status)
survfit(s~1)
survfit(Surv(time, status)~1, data=lung)
survfit(Surv(time, status)~1, data=lung) %>% summary
survfit(s~1)        %>% summary
# ?summary.survfit
range(lung$time)
seq(0, 1100, 100)
survfit(Surv(time, status)~1, data=lung)   %>% summary(times=seq(0, 1100, 100))
survfit(Surv(time, status)~sex, data=lung) %>% summary(times=seq(0, 1100, 100))

# optional - set factor labels
lung$sex
factor(lung$sex, levels=c(1,2), labels=c("male", "female"))
lung$sex <- factor(lung$sex, levels=c(1,2), labels=c("male", "female"))
head(lung, 10)


sfit <- survfit(Surv(time, status)~sex, data=lung)
plot(sfit)
ggsurvplot(sfit)
ggsurvplot(sfit)
ggsurvplot(sfit, conf.int=TRUE, pval=TRUE, risk.table=TRUE)
ggsurvplot(sfit, conf.int=TRUE, pval=TRUE, risk.table=TRUE, 
           palette=c("dodgerblue", "orchid"), risk.table.height=.15)
survdiff(Surv(time, status)~sex, data=lung)

#####
# exercises using built-in colon data
?colon #(click on the chemotherapy data)
colond <- filter(colon, etype==1)
survfit(Surv(time, status)~sex,      data=colond) %>% ggsurvplot(conf.int=TRUE, pval=TRUE)
survfit(Surv(time, status)~rx,       data=colond) %>% ggsurvplot(conf.int=TRUE, pval=TRUE)
survfit(Surv(time, status)~adhere,   data=colond) %>% ggsurvplot(conf.int=TRUE, pval=TRUE)
survfit(Surv(time, status)~differ,   data=colond) %>% ggsurvplot(conf.int=TRUE, pval=TRUE)
survfit(Surv(time, status)~extent,   data=colond) %>% ggsurvplot(conf.int=TRUE, pval=TRUE)
survfit(Surv(time, status)~node4,    data=colond) %>% ggsurvplot(conf.int=TRUE, pval=TRUE)
range(colond$time)
survfit(Surv(time, status)~node4,    data=colond %>% filter(etype==1)) %>% summary(times=seq(0,3500, 500))
#####

# cox ph models
fit <- coxph(Surv(time, status)~sex, data=lung)
fit
fit <- coxph(Surv(time, status)~sex+age+ph.ecog+ph.karno+pat.karno+meal.cal+wt.loss, data=lung)
fit

#####
# exercises using built-in colon data
colond <- filter(colon, etype==1)
coxph(Surv(time, status)~rx, data=colond)
coxph(Surv(time, status)~rx+age+sex+nodes, data=colond)
#####

# Cox ph, and KM curves
# looks like age is significant when not looking at others. Let's make a categorical variable 
# choice of cut point is meaningful!
coxph(Surv(time, status)~age, data=lung)
# can't make a KM curve on a continuous variable.
survfit(Surv(time, status)~age, data=lung) %>% ggsurvplot()
hist(lung$age)
mean(lung$age)
lung$age
cut(lung$age, breaks=c(0, 62, Inf))
cut(lung$age, breaks=c(0, 62, Inf), labels=c("young", "old"))
lung$agecat <- cut(lung$age, breaks=c(0, 62, Inf), labels=c("young", "old"))
survfit(Surv(time, status)~agecat, data=lung) %>% ggsurvplot(pval=TRUE)
lung$agecat <- cut(lung$age, breaks=c(0, 70, Inf), labels=c("young", "old"))
survfit(Surv(time, status)~agecat, data=lung) %>% ggsurvplot(pval=TRUE)
```

## TCGA