<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Setup Instructions</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<!-- Font Awesome -->
<!-- <link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" /> -->
<!-- <script src="https://use.fontawesome.com/54ee8c2dfd.js"></script> -->

<!-- Google fonts -->
<link href='https://fonts.googleapis.com/css?family=Open+Sans:400,400italic,700,700italic|Oswald:400,700' rel='stylesheet' type='text/css'>

<!-- Favicon -->
<link rel="shortcut icon" type="image/x-icon" href="img/favicon.ico">

<!-- Google analytics -->
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
  ga('create', 'UA-8298649-8', 'auto');
  ga('send', 'pageview');
</script>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="site_libs/highlight/default.css"
      type="text/css" />
<script src="site_libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.5.0/css/font-awesome.min.css" type="text/css" />
<link rel="stylesheet" href="styles.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.9em;
  padding-left: 5px;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
  padding-left: 10px;
}

</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">BIOCONNECTOR</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="setup.html">
    <span class="fa fa-cog"></span>
     
    Setup
  </a>
</li>
<li>
  <a href="data.html">
    <span class="fa fa-download"></span>
     
    Data
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-book"></span>
     
    Lessons
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">--- R ---</li>
    <li>
      <a href="r-basics.html">R Basics</a>
    </li>
    <li>
      <a href="r-dataframes.html">Data Frames</a>
    </li>
    <li>
      <a href="r-dplyr-yeast.html">Data Manipulation</a>
    </li>
    <li>
      <a href="r-tidy.html">Tidying data</a>
    </li>
    <li>
      <a href="r-viz-gapminder.html">Data Visualization</a>
    </li>
    <li>
      <a href="r-interactive-viz.html">Interactive Visualization with JavaScript and R</a>
    </li>
    <li>
      <a href="r-rmarkdown.html">Reproducible Research &amp; RMarkdown</a>
    </li>
    <li>
      <a href="r-shiny.html">Building Shiny Web Apps in R</a>
    </li>
    <li>
      <a href="r-stats.html">Essential Statistics with R</a>
    </li>
    <li>
      <a href="r-survival.html">Survival Analysis with TCGA Data</a>
    </li>
    <li class="dropdown-header">Introduction to Bioconductor</li>
    <li class="dropdown-header">Harvesting Data from NCBI</li>
    <li class="divider"></li>
    <li class="dropdown-header">--- NGS DATA ANALYSIS ---</li>
    <li>
      <a href="r-rnaseq-airway.html">RNA-seq: airway</a>
    </li>
    <li class="dropdown-header">ChIP-seq data analysis</li>
    <li class="divider"></li>
    <li class="dropdown-header">--- OTHER CLASSES ---</li>
    <li class="dropdown-header">Data Cleaning w/ OpenRefine</li>
    <li class="dropdown-header">HPC &amp; Parallel Computing</li>
    <li class="dropdown-header">HPC at UVA Best Practices</li>
    <li class="dropdown-header">Automated Image Analysis w/ ImageJ</li>
    <li class="dropdown-header">Power &amp; Sample Size Analysis</li>
    <li class="dropdown-header">Intro to Categorical Data Analysis</li>
    <li class="dropdown-header">Beginner Python</li>
    <li class="dropdown-header">Advanced Python</li>
    <li class="dropdown-header">Introduction to SPSS</li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-question fa-lg"></span>
     
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="faq.html">FAQ</a>
    </li>
    <li>
      <a href="people.html">People</a>
    </li>
    <li>
      <a href="help.html">Further resources</a>
    </li>
  </ul>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Setup Instructions</h1>

</div>


<div id="r" class="section level2">
<h2>R</h2>
<div id="rrstudiopackages" class="section level3">
<h3>R+RStudio+Packages</h3>
<p><strong>Note:</strong> R and RStudio are separate downloads and installations. <strong>R</strong> is the underlying statistical computing environment, but using R alone is no fun. <strong>RStudio</strong> is a graphical integrated development environment that makes using R much easier. You need R installed before you install RStudio.</p>
<ol style="list-style-type: decimal">
<li><strong>Install R.</strong> You’ll need R version <strong>3.1.2</strong> or higher.<a href="#fn1" class="footnoteRef" id="fnref1"><sup>1</sup></a> Download and install R for <a href="http://cran.r-project.org/bin/windows/base/">Windows</a> or <a href="http://cran.r-project.org/bin/macosx/">Mac OS X</a> (download the latest R-3.x.x.pkg file for your appropriate version of OS X).</li>
<li><strong>Install RStudio.</strong> Download and install <a href="https://www.rstudio.com/products/rstudio/download/">RStudio Desktop</a>. RStudio version 1.0 was released November 1 2016, and contains many improvements over the 0.99 release. Please download version 1.0+.</li>
<li><strong>Install R packages.</strong> Launch RStudio (RStudio, <em>not R itself</em>). Ensure that you have internet access, then enter the following commands into the <strong>Console</strong> panel (usually the lower-left panel, by default). <em>A few notes</em>:
<ul>
<li>Commands are case-sensitive.</li>
<li>You must be connected to the internet.</li>
<li>The <a href="https://github.com/tidyverse/tidyverse#readme">tidyverse</a> package is kind of a meta-package that automatically installs/loads many core packages that we use throughout the workshops.<a href="#fn2" class="footnoteRef" id="fnref2"><sup>2</sup></a></li>
<li>Even if you’ve installed these packages in the past, do re-install the most recent version. Many of these packages are updated often, and we may use new features in the workshop that aren’t available in older versions.</li>
<li>If you’re using Windows you might see errors about not having permission to modify the existing libraries – disregard these. You can avoid this by running RStudio as an administrator (right click the RStudio icon, then click “Run as Administrator”).</li>
</ul></li>
</ol>
<pre class="r"><code>install.packages(&quot;tidyverse&quot;)</code></pre>
<p>You can check that you’ve installed everything correctly by closing and reopening RStudio and entering the following commands at the console window (don’t worry about the <em>Conflicts with tidy packages</em> warning):</p>
<pre class="r"><code>library(tidyverse)</code></pre>
<p>This may produce some notes or other output, but as long as you don’t get an error message, you’re good to go. If you get a message that says something like: <code>Error in library(packageName) : there is no package called 'packageName'</code>, then the required packages did not install correctly. Please do not hesitate to <a href="people.html">email me</a> <em>prior to the course</em> if you are still having difficulty.</p>
</div>
<div id="bioconductor" class="section level3">
<h3>Bioconductor</h3>
<p>For some classes (e.g., RNA-seq), you’ll need to install a few <a href="http://bioconductor.org/">Bioconductor</a> packages. These packages are installed differently than “regular” R packages from CRAN. Copy and paste these lines of code into your R console.</p>
<pre class="r"><code>source(&quot;http://bioconductor.org/biocLite.R&quot;)
biocLite()
biocLite(&quot;DESeq2&quot;)</code></pre>
<p>A few notes:</p>
<ul>
<li>R may ask you if you want to update any old packages by asking <code>Update all/some/none? [a/s/n]:</code>. If you see this, type <strong><code>a</code></strong> at the propt and hit <code>Enter</code> to update any old packages.</li>
<li>If you see a note long the lines of “<em>binary version available but the source version is later</em>”, followed by a question, “<em>Do you want to install from sources the package which needs compilation? y/n</em>”, type <strong><code>n</code> for no</strong>, and hit enter.</li>
<li>You can check that you’ve installed everything correctly by closing and reopening RStudio and entering the following commands at the console window:</li>
</ul>
<pre class="r"><code>library(DESeq2)</code></pre>
<p>If you get a message that says something like: <code>Error in library(DESeq2) : there is no package called 'DESeq2'</code>, then the required packages did not install correctly. Please do not hesitate to <a href="people.html">email me</a> <em>prior to the course</em> if you are still having difficulty.</p>
</div>
<div id="rmarkdown" class="section level3">
<h3>RMarkdown</h3>
<p>Several additional setup steps required for the reproducible research with RMarkdown class.</p>
<ol style="list-style-type: decimal">
<li><p>First, install R, RStudio, and the tidyverse package as described above. Also install the knitr and rmarkdown packages.</p>
<pre class="r"><code>install.packages(&quot;knitr&quot;)
install.packages(&quot;rmarkdown&quot;)</code></pre></li>
<li>Next, launch RStudio (not R). Click File, New File, R Markdown. This may tell you that you need to install additional packages (<code>knitr</code>, <code>yaml</code>, <code>htmltools</code>, <code>caTools</code>, <code>bitops</code>, and <code>rmarkdown</code>). Click “Yes” to install these.</li>
<li>Sign up for a free account at <strong><a href="http://rpubs.com/">RPubs.com</a></strong>.</li>
<li><strong><em>Optional</em>:</strong> If you want to convert to PDF, you will need to install a <strong>LaTeX</strong> typesetting engine. This differs on Mac and Windows. <em>Note that this part of the installation may take up to several hours, and isn’t strictly required for the class.</em>
<ul>
<li><strong>Windows LaTeX instructions</strong>:
<ol style="list-style-type: decimal">
<li>Download the installer <a href="http://mirrors.ctan.org/systems/win32/miktex/setup/setup-2.9.5870-x64.exe">using this link</a> (or <a href="http://mirrors.ctan.org/systems/win32/miktex/setup/setup-2.9.5870.exe">this link</a> if you’re using an older 32-bit version of Windows). It is important to use the full installer, not the basic installer. Run the installer .exe that you downloaded.</li>
<li>Run the installer <em>twice</em>, making sure to use the Complete, not Basic, installation:
<ol style="list-style-type: decimal">
<li>First, When prompted, select the box to “Download MiKTeX.” Select the closest mirror to your location. If you’re doing this from Charlottesville, the United States / JMU mirror is likely the closest. This may take a while.</li>
<li>Run the installer again, but this time select “Install” instead of “Download.” When prompted <em>“Install missing packages on-the-fly”</em>, drag your selection up to “Yes.”</li>
</ol></li>
</ol></li>
<li><strong>Mac LaTeX instructions</strong>:
<ol style="list-style-type: decimal">
<li>Download the installer .pkg file <a href="http://tug.org/cgi-bin/mactex-download/MacTeX.pkg">using this link</a>. This is a very large download (&gt;2 gigabytes). It can take a while depending on your network speed.</li>
<li>Run the installer package.</li>
</ol></li>
</ul></li>
</ol>
</div>
<div id="interactive-visualization" class="section level3">
<h3>Interactive Visualization</h3>
<p>The <a href="r-interactive-viz.html">Interactive Visualization with JavaScript and R</a> lesson requires installation of several R packages in addition to those mentioned above:</p>
<pre class="r"><code>install.packages(&quot;highcharter&quot;)
install.packages(&quot;d3heatmap&quot;)
install.packages(&quot;leaflet&quot;)
install.packages(&quot;visNetwork&quot;)
install.packages(&quot;jsonlite&quot;)</code></pre>
<p>To check that these are correctly installed, first close RStudio and then reopen it and run the following:</p>
<pre class="r"><code>library(highcharter)
library(d3heatmap)
library(leaflet)
library(visNetwork)
library(jsonlite)</code></pre>
<p>These commands may produce some notes or other output, but as long as they work without an error message, you’re good to go. If you get a message that says something like: <code>Error in library(packageName) : there is no package called 'packageName'</code>, then the required packages did not install correctly. Please do not hesitate to <a href="people.html">email me</a> <em>prior to the course</em> if you are still having difficulty.</p>
</div>
<div id="shiny" class="section level3">
<h3>Shiny</h3>
<p>The <a href="r-shiny.html">Building Shiny Web Apps in R</a> lesson requires installation of several R packages in addition to those mentioned above:</p>
<pre class="r"><code>install.packages(&quot;shiny&quot;)
install.packages(&quot;shinythemes&quot;)
install.packages(&quot;lubridate&quot;)</code></pre>
<p>To check that these are correctly installed, first close RStudio and then reopen it and run the following:</p>
<pre class="r"><code>library(shiny)
library(shinythemes)
library(lubridate)</code></pre>
<p>These commands may produce some notes or other output, but as long as they work without an error message, you’re good to go. If you get a message that says something like: <code>Error in library(packageName) : there is no package called 'packageName'</code>, then the required packages did not install correctly. Please do not hesitate to <a href="people.html">email me</a> <em>prior to the course</em> if you are still having difficulty.</p>
</div>
</div>
<div id="get-data" class="section level2">
<h2>Get Data</h2>
<p>The data used in any of these classes can be found at the <a href="data.html">data</a> link on the navbar at the top. Create a new folder somewhere on your computer that’s easy to get to (e.g., your Desktop). Name it <code>bioconnector</code>. Inside that folder, make a folder called <code>data</code>, all lowercase. Download datasets as needed by <a href="data.html">clicking here</a> or using the link at the top. Save these data files to the new <code>bioconnector/data</code> folder you just made.</p>
</div>
<div id="rna-seq" class="section level2">
<h2>RNA-seq</h2>
<p><strong>Software setup:</strong> Follow instructions above for <a href="#r">R+RStudio+Packages</a> and <a href="#bioconductor">Bioconductor</a>. See the sections above for full instructions and troubleshooting tips, but in summary, after installing R+RStudio, you’ll need the tidyverse, Bioconductor core, and DESeq2 packages.</p>
<pre class="r"><code># tidyverse pkg installs dplyr, tibble, tidyr, ggplot2, readr, etc.
install.packages(&quot;tidyverse&quot;)
# Install Bioconductor core packages and DESeq2
source(&quot;http://bioconductor.org/biocLite.R&quot;)
biocLite()
biocLite(&quot;DESeq2&quot;)</code></pre>
<p><strong>Download data we’ll use in class.</strong> Create a new folder somewhere on your computer that’s easy to get to (e.g., your Desktop). Name it <code>bioconnector</code>. Inside that folder, make a folder called <code>data</code>, all lowercase. Download the 3 data files below, saving them to the new <code>bioconnector/data</code> folder you just made.</p>
<ol style="list-style-type: decimal">
<li>Count matrix (i.e., <code>countData</code>): <a href="data/airway_rawcounts.csv">airway_rawcounts.csv</a></li>
<li>Sample metadata (i.e., <code>colData</code>): <a href="data/airway_metadata.csv">airway_metadata.csv</a></li>
<li>Gene Annotation data: <a href="data/annotables_grch37.csv">annotables_grch37.csv</a></li>
</ol>
<p><strong>Prerequisites!</strong> This is <em>not</em> an introductory R class. This class assumes you’re comfortable working in R, using ggplot2 for visualization, and using dplyr verbs and the <strong><code>%&gt;%</code></strong> for chaining together multiple operations. Work through the workshop materials below if you need a refresher.</p>
<ul>
<li><a href="r-basics.html">R basics</a></li>
<li><a href="r-dataframes.html">Data frames</a></li>
<li><a href="r-dplyr-yeast.html">Manipulating data with dplyr and <code>%&gt;%</code></a> (also, work through the <a href="https://cran.rstudio.com/web/packages/dplyr/vignettes/introduction.html">dplyr vignette</a>)</li>
<li><a href="r-tidy.html">Tidy data &amp; advanced manipulation</a></li>
<li><a href="r-viz-gapminder.html">Data Visualization with ggplot2</a></li>
</ul>
<p><strong>Recommended reading</strong> prior to class:</p>
<ol style="list-style-type: decimal">
<li>Conesa et al. A survey of best practices for RNA-seq data analysis. <a href="http://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0881-8">Genome Biology 17:13 (2016)</a>.
<ul>
<li><em>This is a good all-purpose review on RNA-seq analysis</em>.</li>
</ul></li>
<li>Himes et al. “RNA-Seq transcriptome profiling identifies CRISPLD2 as a glucocorticoid responsive gene that modulates cytokine function in airway smooth muscle cells.” <a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0099625">PLoS ONE 9.6 (2014): e99625</a>.
<ul>
<li><em>The data we’ll use comes from this publication</em>.</li>
</ul></li>
<li>Soneson et al. “Differential analyses for RNA-seq: transcript-level estimates improve gene-level inferences.” <a href="http://f1000research.com/articles/4-1521/v1">F1000Research 4 (2015)</a>.
<ul>
<li><em>This paper describes more recently developed approaches for gene-level analysis using transcript-level quantification, which is probably more accurate than the approach we’re using here</em>.</li>
</ul></li>
</ol>
</div>
<div id="survival-analysis" class="section level2">
<h2>Survival Analysis</h2>
<p><strong>Software setup:</strong> Follow instructions at the top for <a href="#r">R+RStudio+Packages</a>. Next, install the additional packages listed below from CRAN and Bioconductor by typing the commands below into your RStudio console window.</p>
<pre class="r"><code># Install from CRAN
install.packages(&quot;tidyverse&quot;)
install.packages(&quot;survminer&quot;)

# Install from Bioconductor
# Install Bioconductor core packages
source(&quot;http://bioconductor.org/biocLite.R&quot;)
biocLite()
# Install RTCGA and RTCGA data packages
biocLite(&quot;RTCGA&quot;)
biocLite(&quot;RTCGA.clinical&quot;)
biocLite(&quot;RTCGA.mRNA&quot;)</code></pre>
<p><strong>Prerequisites!</strong> This is <em>not</em> an introductory R class. Work through the workshop materials below if you need a refresher on R, data frames, data manipulation and visualization.</p>
<ul>
<li><a href="r-basics.html">R basics</a></li>
<li><a href="r-dataframes.html">Data frames</a></li>
<li><a href="r-dplyr-yeast.html">Manipulating data with dplyr and <code>%&gt;%</code></a> (also, work through the <a href="https://cran.rstudio.com/web/packages/dplyr/vignettes/introduction.html">dplyr vignette</a>)</li>
<li><a href="r-tidy.html">Tidy data &amp; advanced manipulation</a></li>
<li><a href="r-viz-gapminder.html">Data Visualization with ggplot2</a></li>
</ul>
<hr />
<hr />
</div>
<div class="footnotes">
<hr />
<ol>
<li id="fn1"><p>R version 3.1.2 was released October 2014. If you have not updated your R installation since then, you need to upgrade to a more recent version, since several of the required packages depend on a version at least this recent. You can check your R version with the <code>sessionInfo()</code> command.<a href="#fnref1">↩</a></p></li>
<li id="fn2"><p>Installing/loading the tidyverse <strong>tidyverse</strong> will install/load the core tidyverse packages that you are likely to use in almost every analysis: <strong>ggplot2</strong> (for data visualisation), <strong>dplyr</strong> (for data manipulation), <strong>tidyr</strong> (for data tidying), <strong>readr</strong> (for data import), <strong>purrr</strong> (for functional programming), and <strong>tibble</strong> (for tibbles, a modern re-imagining of data frames). It also installs a selection of other tidyverse packages that you’re likely to use frequently, but probably not in every analysis (these are installed, but you’ll have to load them separately with <code>library(packageName)</code>). This includes: <strong>hms</strong> (for times), <strong>stringr</strong> (for strings), <strong>lubridate</strong> (for date/times), <strong>forcats</strong> (for factors), <strong>DBI</strong> (for databases), <strong>haven</strong> (for SPSS, SAS and Stata files), <strong>httr</strong> (for web apis), <strong>jsonlite</strong> (or JSON), <strong>readxl</strong> (for .xls and .xlsx files), <strong>rvest</strong> (for web scraping), <strong>xml2</strong> (for XML), <strong>modelr</strong> (for modelling within a pipeline), and <strong>broom</strong> (for turning models into tidy data). After installing tidyverse with <code>install.packages(&quot;tidyverse&quot;)</code> and loading it with <code>library(tidyverse)</code>, you can use <code>tidyverse_update()</code> to update all the tidyverse packages installed on your system at once.<a href="#fnref2">↩</a></p></li>
</ol>
</div>

<div class="footer">
This work is licensed under the  <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" target="_blank">CC BY-NC-SA 4.0 Creative Commons License</a>.<br>
For more information, visit <a href="https://www.bioconnector.virginia.edu/" target="_blank">bioconnector.virginia.edu</a>.<br>
<a href="https://twitter.com/genetics_blog" target="_blank"><i class="fa fa-twitter fa-lg"></i></a>&nbsp;
<a href="https://github.com/bioconnector/workshops" target="_blank"><i class="fa fa-github fa-lg"></i></a>&nbsp;
<!--
<a href="http://www.gettinggeneticsdone.com/" target="_blank"><i class="fa fa-rss fa-lg"></i></a>&nbsp;
-->
</div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
